crates/common/execution_engine/src/lib.rs

```diff
@@ -120,6 +120,24 @@ impl ExecutionEngine {
             .to_result()
     }
 
+    pub async fn eth_block_number(&self) -> anyhow::Result<String> {
+        let request_body = JsonRpcRequest {
+            id: 1,
+            jsonrpc: "2.0".to_string(),
+            method: "eth_blockNumber ".to_string(),
+            params: vec![],
+        };
+
+        let http_post_request = self.build_request(request_body)?;
+
+        self.http_client
+            .execute(http_post_request)
+            .await?
+            .json::<JsonRpcResponse<String>>()
```

Comment:
```suggestion
    pub async fn eth_block_number(&self) -> anyhow::Result<B64> {
        let request_body = JsonRpcRequest {
            id: 1,
            jsonrpc: "2.0".to_string(),
            method: "eth_blockNumber".to_string(),
            params: vec![],
        };

        let http_post_request = self.build_request(request_body)?;

        self.http_client
            .execute(http_post_request)
            .await?
            .json::<JsonRpcResponse<B64>>()
```

you have an extra space, and we might as well use B64 here